griptape/structures/structure.py

```diff
@@ -115,18 +122,72 @@ def output_task(self) -> Optional[BaseTask]:
     def finished_tasks(self) -> list[BaseTask]:
         return [s for s in self.tasks if s.is_finished()]
 
+    @property
+    def default_config(self) -> BaseStructureConfig:
+        config = OpenAiStructureConfig()
+
+        if self.prompt_driver is not None:
+            warnings.warn(
+                "`prompt_driver` is deprecated, use `config.global_drivers.prompt_driver` instead.", DeprecationWarning
+            )
+            config.global_drivers.prompt_driver = self.prompt_driver
+            config.task_memory.query_engine.prompt_driver = self.prompt_driver
+            config.task_memory.summary_engine.prompt_driver = self.prompt_driver
+            config.task_memory.extraction_engine.csv.prompt_driver = self.prompt_driver
+            config.task_memory.extraction_engine.json.prompt_driver = self.prompt_driver
+        if self.embedding_driver is not None:
+            warnings.warn(
+                "`embedding_driver` is deprecated, use `config.global_drivers.embedding_driver` instead.",
+                DeprecationWarning,
+            )
+            config.task_memory.query_engine.vector_store_driver.embedding_driver = self.embedding_driver
+        if self.stream is not None:
+            warnings.warn("`stream` is deprecated, use `config.prompt_driver.stream` instead.", DeprecationWarning)
+            config.global_drivers.prompt_driver.stream = self.stream
+
+        return config
+
     @property
     def default_task_memory(self) -> TaskMemory:
+        global_drivers = self.config.global_drivers
+        task_memory = self.config.task_memory
+
         return TaskMemory(
             artifact_storages={
                 TextArtifact: TextArtifactStorage(
                     query_engine=VectorQueryEngine(
-                        prompt_driver=self.prompt_driver,
-                        vector_store_driver=LocalVectorStoreDriver(embedding_driver=self.embedding_driver),
+                        prompt_driver=(
+                            task_memory.query_engine.prompt_driver
+                            if not isinstance(task_memory.query_engine.prompt_driver, NopPromptDriver)
+                            else global_drivers.prompt_driver
```

Comment:
Can we flip it here and below to not use a negative condition, so it's easier to read and reason about?

Reply:
Done